internal/mesh/internal/controllers/sidecar-proxy/controller.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package sidecar_proxy
```

Comment:
(nit)

Do we want to have a package with a native underscore, vs `sidecarproxy`?

Reply:
renamed!